runtime/lua/vim/treesitter/query.lua

```diff
@@ -55,13 +58,26 @@ function M.get_query_files(lang, query_name, is_included)
   --
   -- {language} ::= {lang} | ({lang})
   local MODELINE_FORMAT = '^;+%s*inherits%s*:?%s*([a-z_,()]+)%s*$'
+  local EXTENDS_FORMAT = '^;+ extends$'
 
-  for _, file in ipairs(lang_files) do
-    local modeline = safe_read(file, '*l')
+  for _, filename in ipairs(lang_files) do
+    local file, err = io.open(filename, 'r')
+    if not file then
+      error(err)
+    end
 
-    if modeline then
-      local langlist = modeline:match(MODELINE_FORMAT)
+    local extension = false
+
+    for modeline in
+      function()
+        return file:read('*l')
```

Comment:
Why are you not using `safe_read` any more? That should handle all the boilerplate here and above.

Reply:
Because I don't know how much lines I would like to read in advance.
This way I can just open the file once (beginning of the loop), iterate it's lines, and then close it.

Reply:
Makes sense. But in this case we don't really need the `safe_read` function anymore and can just do the same in `read_query_files` below?

Reply:
I'd suggest keeping it, we may need it at some later point.
And it does not hurt to keep ?

Reply:
Sure, just a thought.